ompi/proc/proc.h

```diff
@@ -369,16 +369,42 @@ static inline bool ompi_proc_is_sentinel (ompi_proc_t *proc)
     return (intptr_t) proc & 0x1;
 }
 
-static inline intptr_t ompi_proc_name_to_sentinel (opal_process_name_t name)
+/*
+ * we assume an ompi_proc_t is at least aligned on two bytes,
+ * so if the LSB of a pointer to an ompi_proc_t is 1, we have to handle
+ * this pointer as a sentinel instead of a pointer.
+ * a sentinel can be seen as a 64 bits array with the following format :
+ * - bit  0     : 1
+ * - bits 1-15  : local jobid
+ * - bits 16-31 : job family
+ * - bits 32-63 : vpid
```

Comment:
It sounds like a really fragile idea to me, and unnecessarily breaks 32-bit environments. Why not just have it always be a pointer to at least a process name? In other words, always create the opal_process_name_t portion of the ompi_proc_t, and then just embed that in an ompi_proc_t when you fill it in? I understand it costs us a touch more memory, but at least it wouldn't be so vulnerable to any future changes in opal_process_name_t.

We made the mistake in the past of trying to play bit games with process names, and got burned for it. Seems like we might be heading down that path again.

Anyway, that's my $0.02.